plugins/vli/fu-vli-pd-firmware.c

```diff
@@ -79,28 +61,6 @@ fu_vli_pd_firmware_parse(FuFirmware *firmware,
 	if (!fu_input_stream_size(stream, &streamsz, error))
 		return FALSE;
 
-	/* fall back to legacy location */
-	if (!fu_vli_pd_firmware_validate_header(self)) {
-		g_byte_array_unref(st);
-		st = fu_struct_vli_pd_hdr_parse_stream(stream,
-						       VLI_USBHUB_PD_FLASHMAP_ADDR_LEGACY,
-						       error);
-		if (st == NULL) {
-			g_prefix_error(error, "failed to read header: ");
-			return FALSE;
-		}
-	}
-	self->vid = fu_struct_vli_pd_hdr_get_vid(st);
```

Comment:
I think there's more places that reference self->vid that we should delete too. e.g. i think we want to remove `guint16 vid` and `guint16 pid` from `struct _FuVliPdFirmware`

Reply:
Yes, and should I also remove fu_xml_bilder_insert_kx(bn, "vid", self->vid); from fu_vli_pd_firmware_export() ?

Reply:
I think so.